apps/block_scout_web/lib/block_scout_web/controllers/forward_transfer_controller.ex

```diff
@@ -16,9 +17,25 @@ defmodule BlockScoutWeb.ForwardTransferController do
 
   alias Explorer.{Chain}
 
+  @necessity_by_association %{
+    :block => :optional,
+    [to_address: :names] => :optional,
+    [to_address: :smart_contract] => :optional
+  }
+
+  @default_options [
+    necessity_by_association: %{
+      :block => :required,
+      [to_address: :names] => :optional,
+      [to_address: :smart_contract] => :optional
+    }
+  ]
+
```

Comment:
I asked on the [other PR](https://github.com/HorizenLabs/sc_evm_blockscout/pull/167/files), but leaving the questions here too: what is this doing exactly? could you help me understand the necessity_by_association statements? I see it's used on fetch_recent_collated_forward_transfers_for_rap method but I'm not sure about it.

Reply:
sure thing, @jmontesdeocanuez  forward transfer records are being fetched from the forward_transfers table and there are also associated records like block and address that can fetched as well. these lines specify what associated data have to be fetched and which should just be 'pre-loaded'. associated blocks have to be fetched, associated addresses need to be pre-loaded, which i think means the reference is fetched but not the full object itself